db/migrations/mysql/000029_create_sidebar_categories.up.sql

```diff
@@ -0,0 +1,57 @@
+CREATE TABLE IF NOT EXISTS SidebarCategories (
+    Id varchar(128) NOT NULL,
+    UserId varchar(26) DEFAULT NULL,
+    TeamId varchar(26) DEFAULT NULL,
+    SortOrder bigint(20) DEFAULT NULL,
+    Sorting varchar(64) DEFAULT NULL,
+    Type varchar(64) DEFAULT NULL,
+    DisplayName varchar(64) DEFAULT NULL,
+    Muted tinyint(1) DEFAULT 0,
+    Collapsed tinyint(1) DEFAULT 0,
+    PRIMARY KEY (Id)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;
+
+SET @preparedStatement = (SELECT IF(
+    (
+        SELECT COUNT(*) FROM INFORMATION_SCHEMA.COLUMNS
+        WHERE table_name = 'SidebarCategories'
+        AND table_schema = DATABASE()
+        AND column_name = 'Id'
```

Comment:
Since this is a modification, I know that previously we were doing `UpdateColumnTypeIfNotExists` but in the "new" migrations we have the capabilities to use a more granular control and check if the column with that specific type exists. Something like:

```sql
AND column_type != 'varchar(128)'
```

Reply:
TIL 👍